wrapped-bridged-usdc/src/lib.rs

```diff
@@ -57,6 +69,43 @@ pub trait WrappedBridgedUsdc {
         );
     }
 
+    /// Will wrap what it can, and send back the rest unchanged
+    #[payable("*")]
+    #[endpoint(wrapMultipleTokens)]
+    fn wrap_multiple_tokens(&self) -> PaymentsVec<Self::Api> {
+        let original_payments = self.call_value().all_esdt_transfers();
+        if original_payments.is_empty() {
+            return original_payments;
+        }
+
+        let mut new_payments = ManagedVec::new();
+        let token_whitelist = self.chain_specific_usdc_token_ids();
+        let universal_token_id = self.universal_bridged_usdc_token_id().get();
+
+        for p in &original_payments {
+            let new_payment = if token_whitelist.contains(&p.token_identifier) {
+                self.send()
+                    .esdt_local_mint(&universal_token_id, 0, &p.amount);
+
+                EsdtTokenPayment {
+                    token_type: EsdtTokenType::Fungible,
+                    token_identifier: universal_token_id.clone(),
+                    token_nonce: 0,
+                    amount: p.amount,
+                }
+            } else {
+                p
```

Comment:
understand the logic - and that it is easier to code like this - but I do not see this as expandable as possible - I think here you should return error if token is not whitelisted. For the long term, when we have multiple tokens to wrap - it will be a lot of gas.

Reply:
Returning error here ruins the whole logic. The whole idea is that the MultiTransfer SC sends everything to the wrapping contract, so we don't have to duplicate whitelists. Anything that's not whitelisted means it doesn't need wrapping.